guides/plugins/apps/administration/flow-builder/add-custom-flow-actions-from-app-system.md

```diff
@@ -0,0 +1,255 @@
+
+```xml
+<flow-action>
+    <meta>
+        <name>slack.send.message</name>
+        <label>Send slack message</label>
+        <label lang="de-DE">Send slack message</label>
+        <badge>Slack</badge>
+        <description>Slack send message description</description>
+        <description lang="de-DE">Slack send message description DE</description>
+        <url>https://hooks.slack.com/services/{id}</url>
+        <sw-icon>default-communication-speech-bubbles</sw-icon>
+        <icon>slack.png</icon>
+        <requirements>orderAware</requirements>
+        <requirements>customerAware</requirements>
+    </meta>
+    ...
+</flow-action>
+```
+
+| Key | Required | Description |
+| :--- | :--- | :--- |
+| name | yes | The technical name of your action, unique for all actions |
+| label | yes | A name to be shown for your action in the actions list or action modal title |
+| badge | no | An attached badge shown behind the label in the action modal title |
+| description | yes | Detailed information for your action |
+| sw-icon | no | An icon component name from the [icon library](https://component-library.shopware.com/icons/) |
+| icon | no | Alternatively, a path to your action icon |
+| requirements | yes | Available action triggers, read more below |
+| url | yes | External webhook location. Shopware will call this URL when the action is executed |
+
+**requirements**
+
+Requirements will decide for which trigger events you action is available.
+Example: The `checkout.order.placed` has an `orderAware` requirements - indicating that your action is allow to use be used in the `checkout.order.placed` event. It is defined using `<requirements>orderAware</requirements>` in your app action definition.
+
+### Header parameters
+
+```xml
+<flow-action>
+    <meta>
+        ...
+    </meta>
+    <headers>
+        <parameter type="string" name="content-type" value="application/json"/>
+    </headers>
+    ...
+<flow-action>
+ ```
+
+| Key | Description |
+| :--- | :--- |
+| type | Parameter type - currently only `string` supported |
+| name | The header key |
+| value | The header value |
+
+### Parameters
+
+```xml
+<flow-action>
+    <meta>
+        ...
+    </meta>
+    <headers>
+        ...
+    </headers>
+    <parameters>
+        <parameter type="string" name="text" value="{{ message }} \n Order Number: {{ order.orderNumber }}"/>
+    </parameters>
+    ...
+<flow-action>
+ ```
+
+Define the `parameter` for the URL body based on your URL webhook services.
+
+| Key | Description |
+| :--- | :--- |
+| type | Type of parameter, only support `string` type. |
+| name | The body key for your URL. |
+| value | The content message for your URL; free to design your content message here. |
+| {{ message }} | The variable from your `<input-field>` defined in `flow-action.xml`. |
+| {{ order.orderNumber }} | For each trigger event, the action will have the variables suitable. In this case, the `checkout.order.placed` will have `order` variable. |
```

Comment:
Not just the `checkout.order.placed` event, but all events that are order aware ;) 

with the requirements you can specify which data you need to access in the action configuration and you ensure that that data is available in the configured flow trigger event